### PR TITLE
icon: implement aggregator contract

### PR DESCRIPTION
**Pull request by prin-r**
## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [x] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

_________________

std basic contract: https://bicon.tracker.solidwallet.io/contract/cxc79c2120992356eac409d5cf5ff650f2780a6995#readcontract

std proxy contract: https://bicon.tracker.solidwallet.io/contract/cx61a36e5d10412e03c907a507d1e8c6c3856d9964#readcontract

### CODE REVIEW

File: bridges/icon/std_reference_basic/std_reference_basic.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    ONE = 1_000_000_000
+    MICRO_SEC = 1_000_000
+
+    @eventlog(indexed=1)
+    def RefDataUpdate(self, _symbol: str, _rate: int, _last_update: int, _request_id: int):
+        pass
+
+    def __init__(self, db: IconScoreDatabase) -> None:
+        super().__init__(db)
+        # Mapping from symbol to obi encoded ref data.
```
Comment (sorawit) [inline bridges/icon/std_reference_basic/std_reference_basic.py]: invalid comment?

File: bridges/icon/std_reference_basic/std_reference_basic.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+TAG = "StdReferenceBasic"
+
+
+class StdReferenceBasic(IconScoreBase):
+
+    ONE = 1_000_000_000
+    MICRO_SEC = 1_000_000
+
+    @eventlog(indexed=1)
+    def RefDataUpdate(self, _symbol: str, _rate: int, _last_update: int, _request_id: int):
```
Comment (sorawit) [inline bridges/icon/std_reference_basic/std_reference_basic.py:12]: do not index

**Review by sorawit — CHANGES_REQUESTED** [review]

File: bridges/icon/std_reference_basic/std_reference_basic.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    def _get_ref_data(self, _symbol: str) -> dict:
+        if _symbol == "USD":
+            return {"rate": self.ONE, "last_update": self.block.timestamp, "request_id": 0}
+
+        if self.refs[_symbol] == "":
+            self.revert("REF_DATA_NOT_AVAILABLE")
+
+        return json_loads(self.refs[_symbol])
+
+    def _get_reference_data(self, _pair: str) -> dict:
```
Comment (prin-r) [inline bridges/icon/std_reference_basic/std_reference_basic.py]: fix the param according to the specification

File: bridges/icon/std_reference_basic/std_reference_basic.py (hunk 1 above)
Comment (prin-r) [reply bridges/icon/std_reference_basic/std_reference_basic.py]: done

**Review by sorawit — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**